### PR TITLE
Layer factory refactoring

### PR DESCRIPTION
**Pull request by elemoine**
This PR is a refactoring of the layer factory code. There are now more services. And the `appGetWmtsLayer` service is now used in the main controller to create the background layer. Later, the `appGetWmtsLayer` will be used by the background layer component.

Please review.

### CODE REVIEW

File: geoportailv3/static/js/layerfactoryservice.js [hunk 1]
```diff
@@ -9,116 +12,164 @@ goog.require('ol.tilegrid.WMTS');
...
+    var type = node['type'];
+    var layerCacheKey = type + '_' + node['name'];
+    if (layerCacheKey in app.layerCache_) {
+      return app.layerCache_[layerCacheKey];
+    }
+
+    var layer;
+
+    if (type.indexOf('WMS') != -1) {
+      layer = appGetWmsLayer(node['name'], node['name'], node['url']);
```
Comment (pgiraud) [inline geoportailv3/static/js/layerfactoryservice.js]: I don't see any reason why `name` and `label` would be different.

**Comment by pgiraud** [comment]
I added one comment. Otherwise, looks good to me.

File: geoportailv3/static/js/layerfactoryservice.js (hunk 1 above)
Comment (elemoine) [inline geoportailv3/static/js/layerfactoryservice.js]: You're right. I removed the `label` argument.

**Comment by elemoine** [comment]
For the record, I will need to refactor the layer factory services again for the background layer component, but this is a start, and this also fixes a few bugs I've seen while doing the refactoring.

**Pull request merged**